test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilStateChange.java

```diff
@@ -55,4 +69,30 @@ public static List<ContractStateChange> stateChangesToGrpc(List<StateChange> sta
 
 		return additions;
 	}
+
+	public static void initializeEthereumAccountForSpec(final HapiApiSpec spec) {
+		final var newSpecKey = new NewSpecKey(secp256k1SourceKey).shape(secp256k1Shape);
+		final var cryptoTransfer = new HapiCryptoTransfer(HapiCryptoTransfer.tinyBarsFromAccountToAlias(GENESIS, secp256k1SourceKey, ONE_HUNDRED_HBARS));
+
+		newSpecKey.execFor(spec);
+		cryptoTransfer.execFor(spec);
+
+		specToInitializedEthereumContract.putIfAbsent(spec.getName(), true);
+	}
+
+	public static boolean isEthereumAccountCreatedForSpec(final String spec) {
+		return specToInitializedEthereumContract.containsKey(spec);
+	}
+
+	public static void markSpecAsBeenExecuted(final String spec) {
+		specToBeenExecuted.putIfAbsent(spec, true);
+	}
+
+	public static boolean hasSpecBeenExecuted(final String spec) {
+		if(specToBeenExecuted.containsKey(spec)) {
+			return specToBeenExecuted.get(spec);
+		} else {
+			return false;
+		}
```

Comment:
nit: can remove the else block and directly return false after the if has been skipped.